File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.java
import java.util.*;

class Solution {
    public String solution(String[] participant, String[] completion) {
        String answer = "";
        
        HashMap<String, Integer> hashMap = new HashMap<>();
        
        for(String p: participant){
            hashMap.put(p, hashMap.getOrDefault(p, 0)+1);
        }
        
        for(String c: completion){
            hashMap.put(c, hashMap.get(c)-1);
        }
        
        for(String k: hashMap.keySet()){
            if(hashMap.get(k)>0){
                answer = k;
                break;
            }
        }
        
        return answer;
    }
}

File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.py
def solution(participant, completion):
    answer = ''
    
    runner = {}
    
    for i in range(len(participant)):
        runner[participant[i]] = runner.setdefault(participant[i], 0) + 1
    
    for c in completion:
        runner[c] -= 1
    
    for p in participant:
        if runner[p] == 1:
            answer = p
    
    return answer